src/main/java/org/peergos/protocol/dht/KademliaEngine.java

```diff
@@ -46,7 +47,7 @@ public void addIncomingConnection(PeerId peer, Multiaddr addr) {
     public List<PeerAddresses> getKClosestPeers(byte[] key) {
         int k = 20;
         List<Node> nodes = router.find(Id.create(Hash.sha256(key), 256), k, false);
-        System.out.println("Nodes: " + nodes.size());
+        LOG.fine("Nodes: " + nodes.size());
```

Comment:
This log can be removed or maybe changed to "X Nodes discovered by router". What do you think?

Reply:
I don't feel like it conveys the right message as it is.

Reply:
You're right